File: java/junit5/src/main/java/Board.java
public class Board {

  private Cell cells[][];
  private int width;
  private int height;
  public Board(int width, int height){
    this.height = height;
    this.width = width;
    cells = new Cell[width][height];
  }
  public Cell[][] getCells(){
    return cells;
  }
  public int getWidth(){
    return width;
  }
  public int getHeight(){
    return getHeight();
  }

  public Cell getCell(int height, int width){
    return cells[height][width];
  }
}

File: php/src/Board.php
<?php

declare(strict_types=1);

namespace swkberlin;

class Board
{
    private int $width;
    private int $height;
    private array $grid;

    public function __construct(int $width, int $height)
    {
        $this->width = $width;
        $this->height = $height;
        $this->grid = array_fill(0, $height, array_fill(0, $width, ' '));
    }

    public function isWithinBounds(Position $position): bool
    {
        return $position->getX() >= 0 && $position->getX() < $this->width &&
               $position->getY() >= 0 && $position->getY() < $this->height;
    }

    public function getWidth(): int
    {
        return $this->width;
    }

    public function getHeight(): int
    {
        return $this->height;
    }

    public function getGrid(): array
    {
        return $this->grid;
    }

    public function setGridValue(Position $position, string $value): void
    {
        $this->grid[$position->getY()][$position->getX()] = $value;
    }

    public function getGridValue(Position $position): string
    {
        return $this->grid[$position->getY()][$position->getX()];
    }
}